[Inductor][Tritonparse] Call `jit_post_compile_hook` within Inductor Triton Kernel compile path

NikhilAPatel opened the pull request:
Summary: Since Inductor skips JIT compilation for Triton kernels, we need to manually invoke `knobs.runtime.jit_post_compile_hook` if one exists. Here, we do this to enable Tritonparse to extract launch metadata from Inductor launched kernels. We can control whether or not Inductor will run the hook with a new `TORCHINDUCTOR_RUN_JIT_POST_COMPILE_HOOK=1 ` config variable.

Reviewed By: davidberard98

Differential Revision: D80624932




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

FindHao reviewed (approved):
LGTM! please fix the lint error before land.

NikhilAPatel committed:
[Inductor][Tritonparse] Call `jit_post_compile_hook` within Inductor Triton Kernel compile path (#161443)

Summary:
X-link: https://github.com/meta-pytorch/tritonparse/pull/83

Pull Request resolved: https://github.com/pytorch/pytorch/pull/161443

Since Inductor skips JIT compilation for Triton kernels, we need to manually invoke `knobs.runtime.jit_post_compile_hook` if one exists. Here, we do this to enable Tritonparse to extract launch metadata from Inductor launched kernels. We can control whether or not Inductor will run the hook with a new `TORCHINDUCTOR_RUN_JIT_POST_COMPILE_HOOK=1 ` config variable.

Reviewed By: davidberard98, FindHao

Differential Revision: D80624932
